core/iwasm/include/wasm_export.h

```diff
@@ -1546,6 +1546,47 @@ wasm_runtime_set_context_spread(wasm_module_inst_t inst, void *key,
 WASM_RUNTIME_API_EXTERN void *
 wasm_runtime_get_context(wasm_module_inst_t inst, void *key);
 
+/*
+ * wasm_runtime_begin_blocking_op/wasm_runtime_end_blocking_op
+ *
+ * These APIs are intended to be used by the implementations of
+ * host functions. It wraps an operation which possibly blocks for long
+ * to prepare for async termination.
+ *
+ * eg.
+ *
+ *   if (!wasm_runtime_begin_blocking_op) {
+ *       return EINTR;
+ *   }
+ *   ret = possibly_blocking_op();
+ *   wasm_runtime_end_blocking_op(env);
```

Comment:
Had better `wasm_runtime_end_blocking_op(exec_env)`?

Reply:
Should we mention that two APIs take effect only when thread manager is enabled?

Reply:
sure. done.